src/sipsess/accept.c

```diff
@@ -74,8 +74,8 @@ int sipsess_accept(struct sipsess **sessp, struct sipsess_sock *sock,
 	va_list ap;
 	int err;
 
-	if (!sessp || !sock || !msg || scode < 101 || scode > 299 ||
-	    !cuser || !ctype)
+	if (!sessp || !sock || !msg || (scode != 0 && scode < 100) ||
+	    scode > 299 || !cuser || !ctype)
```

Comment:
I am not sure with this. Why not directly use `sipsess_alloc()`?

Reply:
I'll  investigate.

Reply:
`sipsess_alloc()` is not in re API.

Reply:
@cspiel1 What is wrong with `sipsess_accept()`? In addition to `sipsess_alloc()` it does lots other stuff that would also need to be handled separately:
```
err = sip_dialog_accept(&sess->dlg, msg);
	if (err)
		goto out;

	hash_append(sock->ht_sess,
		    hash_joaat_str(sip_dialog_callid(sess->dlg)),
		    &sess->he, sess);

	sess->msg = mem_ref((void *)msg);

	err = sip_strans_alloc(&sess->st, sess->sip, msg, cancel_handler,
			       sess);
```
So I would prefer to keep the PR as it currently is.  I have made tests and they all have worked fine.

Reply:
In my opinion `sipsess_accept()` should send a reply. @alfredh and @sreimers please review and decide!

Should we revert this, when we have another solution?